Examples/Tests/PythonWrappers/PICMI_inputs_2d.py

```diff
@@ -0,0 +1,262 @@
+    Expml = pwxf.ExFPPMLWrapper(include_ghosts = include_ghosts)
+    Eypml = pwxf.EyFPPMLWrapper(include_ghosts = include_ghosts)
+    Ezpml = pwxf.EzFPPMLWrapper(include_ghosts = include_ghosts)
+    Bxpml = pwxf.BxFPPMLWrapper(include_ghosts = include_ghosts)
+    Bypml = pwxf.ByFPPMLWrapper(include_ghosts = include_ghosts)
+    Bzpml = pwxf.BzFPPMLWrapper(include_ghosts = include_ghosts)
+    Fpml  = pwxf.FFPPMLWrapper(include_ghosts = include_ghosts)
+    Gpml  = pwxf.GFPPMLWrapper(include_ghosts = include_ghosts)
+
+# Compute min and max of fields data
+def compute_minmax(data):
+    vmin = -max(abs(data.min()), abs(data.max()))
+    vmax =  max(abs(data.min()), abs(data.max()))
+    return (vmin, vmax)
+
+# Plot fields data either in valid domain or in PML
+def plot_data(it, data, edge, pml, title, name):
+    fig, ax = plt.subplots(nrows = 1, ncols = 1, gridspec_kw = dict(wspace = 0.5), figsize = [6,5])
+    cax = make_axes_locatable(ax).append_axes('right', size='5%', pad='5%')
+    vmin, vmax = compute_minmax(data)
+    im = ax.imshow(data, origin = 'lower', extent = edge, vmin = vmin, vmax = vmax, cmap = 'seismic')
+    ax.set_xlabel('x')
+    ax.set_ylabel('z')
+    ax.set_title(title)
+    fig.colorbar(im, cax = cax)
+    # Draw PMLs and ghost regions
+    lw  = 0.8
+    ls = '-'
+    if (pml):
+        ax.axvline(x = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = 0+nxg   , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axvline(x = -nxpml  , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx-nxg  , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axvline(x = nx+nxpml, linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0+nzg   , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axhline(y = -nzpml  , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz-nzg  , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axhline(y = nz+nzpml, linewidth = lw, color = 'grey' , linestyle = ls)
+    else:
+        ax.axvline(x = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz      , linewidth = lw, color = 'grey' , linestyle = ls)
+    # Plot title
+    suptitle = 'PML in (x,z), 4 grids 64 x 64'
+    plt.suptitle(suptitle)
+    figname = 'figure_' + name + '.png'
+    fig.savefig(figname, dpi = 100)
+
+# Initialize fields data (unit pulse) and apply smoothing
+def init_data(data):
+    data[nx//2,nz//2] = 1.
+    data[2:nx-1,:] *= 0.5
```

Comment:
Doing the smoothing over the full grid seems excessive. I think this code would be cleaner if you write out the impulse matrix and copy it to the array. For example

> impulse = np.array([1./16., 1./8.. 1/16.], [1./8., 1./4., 1./8.], [1./16., 1./8.. 1/16.]])
data[nx//2-1:nx//2+2, ny//2-1:ny//2+2] = impulse

Or something fancier

> impulse1d = np.array([1./4., 1./2., 1./4.])
impulse = np.outer(impulse1d, impulse1d)
data[nx//2-1:nx//2+2, ny//2-1:ny//2+2] = impulse

Reply:
Thank you for this suggestion, fixed in https://github.com/ECP-WarpX/WarpX/pull/2576/commits/b05e0e1814c69c7b85c78ec6202eaef6bb08de81.